holmes/core/tool_calling_llm.py

```
@@ -909,15 +1037,86 @@ def call_stream(
 
                 for future in concurrent.futures.as_completed(futures):
                     tool_call_result: ToolCallResult = future.result()
-                    tool_calls.append(tool_call_result.as_tool_result_response())
-                    messages.append(tool_call_result.as_tool_call_message())
 
-                    perf_timing.measure(f"tool completed {tool_call_result.tool_name}")
+                    if (
+                        tool_call_result.result.status
+                        == ToolResultStatus.APPROVAL_REQUIRED
```

Ruff (0.12.2)

Undefined name `ToolResultStatus`

(F821)